impl-base/src/main/java/org/jboss/shrinkwrap/impl/base/URLPackageScanner.java

```diff
@@ -156,7 +156,8 @@ private void handle(File file, String packageName) throws ClassNotFoundException
                 String className = packagePrefix + child.getName().substring(0, child.getName().lastIndexOf(SUFFIX_CLASS));
                 foundClass(className, prefix + className.replace( '.', '/' ) + SUFFIX_CLASS );
             } else if (child.isDirectory() && addRecursively) {
-                handle(child, packageName + "." + child.getName());
+                final String nextPackageName = packageName.length() > 0 ? packageName + "." + child.getName() : child.getName();
```

Comment:
In the meantime I have created a very simple test demonstrating the issue but due to classloader magic etc. (?) in other tests it fails with:
```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.145 sec <<< FAILURE! - in org.jboss.shrinkwrap.impl.base.spec.SimpleAddPackageTestCase
testAddPackageFromMainToJar(org.jboss.shrinkwrap.impl.base.spec.SimpleAddPackageTestCase)  Time elapsed: 0.143 sec  <<< ERROR!
java.lang.IllegalArgumentException: exploded_import_test/arbitraryDirectory/test/war/WEB-INF/classes/org/jboss/shrinkwrap/api/importer/ExplodedImporter.class not found in classloader sun.misc.Launcher$AppClassLoader@133056f
        at org.jboss.shrinkwrap.api.asset.ClassLoaderAsset.<init>(ClassLoaderAsset.java:70)
        at org.jboss.shrinkwrap.impl.base.URLPackageScanner.foundClass(URLPackageScanner.java:167)
        at ...
```
There is no failure when running the test separately via `-Dtest=...`.